### PR TITLE
Improved TWAP slice amount validation

### PR DESCRIPTION
**Pull request by alexstotsky**
### PR description
- Added validation for cases when slice amount greater than amount
- Related to https://app.asana.com/0/1125859137800433/1200016200363066

slice-am-valid-err

### CODE REVIEW

File: src/components/OrderForm/OrderForm.js [hunk 1]
```diff
@@ -403,6 +403,8 @@ export default class OrderForm extends React.Component {
     const renderData = marketToQuoteBase(currentMarket)
     const atomicOrderTypes = []
     const algoOrderTypes = []
+    const { amount, sliceAmount } = fieldData
+    if (+sliceAmount > +amount) validationErrors.sliceAmount = 'Slice amount cannot be greater than amount'
```
Comment (robertkowalski) [inline src/components/OrderForm/OrderForm.js:407]: adding a specific fix for twap orders in the general order form component feels a bit strange.

we also require the lib `bfx-hf-algo` in the ui, where the check_param functionality should be available

how about a more general approach where the validation from the module at https://github.com/bitfinexcom/bfx-hf-algo/blob/ad890273da217362483c84505394af52b4d6ada7/lib/twap/meta/validate_params.js#L45 is used here, similar to the ui definitions that are used?

Comment (robertkowalski) [reply src/components/OrderForm/OrderForm.js:407]: maybe this here is just failing?

https://github.com/bitfinexcom/bfx-hf-ui/blob/e8278a164023bdb634d42187f43c4d3783afb098/src/components/OrderForm/OrderForm.js#L219-L220

Comment (alexstotsky) [reply src/components/OrderForm/OrderForm.js:407]: ok I'll check if we can implement it this way

Comment (robertkowalski) [reply src/components/OrderForm/OrderForm.js:407]: thanks @alexstotsky -- it seems there is already a check, but it does not work any more?

https://github.com/bitfinexcom/bfx-hf-ui/blob/e8278a164023bdb634d42187f43c4d3783afb098/src/components/OrderForm/OrderForm.js#L219-L220

**Comment by alexstotsky** [comment]
`OrderForm` fields validation will be rebuilt the way you proposed in a separate PR and related to all order types and fields, not just for TWAP and slice amount. This one will be closed.

**Pull request closed without merging**